src/com/owncloud/android/authentication/AuthenticatorActivity.java

```diff
@@ -987,35 +1071,43 @@ private void startOauthorization() {
         OAuth2Provider oAuth2Provider = OAuth2ProvidersRegistry.getInstance().getProvider();
         oAuth2Provider.setAuthorizationServerUri(mServerInfo.mBaseUrl);
 
-        OAuth2RequestBuilder builder = oAuth2Provider.getOperationBuilder();
+        OAuth2RequestBuilder builder =
+                oAuth2Provider.getOperationBuilder();
         builder.setGrantType(OAuth2GrantType.AUTHORIZATION_CODE);
         builder.setRequest(OAuth2RequestBuilder.OAuthRequest.GET_AUTHORIZATION_CODE);
 
-        String authorizationCodeUri = builder.buildUri();
+        if(mCustomTabPackageName != null) {
+            openUrlWithCustomTab(builder.buildUri());
+        } else {
+            openUrlInBrowser(builder.buildUri());
+        }
+    }
 
-        // Show OAuth web dialog
-        ArrayList<String> targetUrls = new ArrayList<>();
-        targetUrls.add(getString(R.string.oauth2_redirect_uri));    // target URI for successful authorization
-        targetUrls.add(mServerInfo.mBaseUrl + OwnCloudClient.FILES_WEB_PATH);
+    private void openUrlWithCustomTab(String url) {
+        TypedValue tvBackground = new TypedValue();

```

Comment:
Not sure if tvBackground is a descriptive or a good name for this variable, what do you mean? Isn't it anything related to 📺 , is it?

Reply:
yea probably better to change ... copypasta error